ui/app/src/components/CreateDashboardDialog/CreateDashboardInProjectDialog.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useCallback, useState } from 'react';
+import { Button, FormControl, InputLabel, MenuItem, Select, SelectChangeEvent, TextField } from '@mui/material';
+import { Dialog } from '@perses-dev/components';
+import { DashboardSelector } from '@perses-dev/core';
+
+export interface CreateDashboardInProjectProps {
+  open: boolean;
+  projectOptions: string[];
+  onClose: DispatchWithoutAction;
+  onSuccess?: Dispatch<DashboardSelector>;
+}
+
+/**
+ * Dialog used to create a dashboard.
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @param props.projectName The project where the dashboard will be created.
+ * @constructor
+ */
+export const CreateDashboardInProjectDialog = (props: CreateDashboardInProjectProps) => {
```

Comment:
We already have a `CreateDashboardDialog`. Is it possible to refactor that dialog so it works here as well?